src/internal/Subscription.ts

```diff
@@ -18,20 +19,27 @@ import { SubscriptionLike, TeardownLogic } from './types';
  */
 export class Subscription implements SubscriptionLike {
   /** @nocollapse */
-  public static EMPTY: Subscription = (function(empty: any) {
+  public static EMPTY: Subscription = (function (empty: any) {
     empty.closed = true;
     return empty;
-  }(new Subscription()));
+  })(new Subscription());
 
   /**
    * A flag to indicate whether this Subscription has already been unsubscribed.
    */
   public closed = false;
 
-  /** @internal */
-  protected _parentOrParents: Subscription | Subscription[] | null = null;
-  /** @internal */
-  private _subscriptions: SubscriptionLike[] | null = null;
+  /** If this subscription has been added to one parent, it will show up here */
+  private _singleParent?: Subscription;
+
+  /** If this subscription has been added to more than one parent, they will show up here. */
+  private _parents?: Subscription[];
+
+  /**
+   * The list of registered teardowns to execute upon unsubscription. Adding and removing from this
+   * list occurs in the {@link add} and {@link remove} methods.
+   */
+  private _teardowns?: Exclude<TeardownLogic, void>[];
```

Comment:
question: Is there a particular reason for not initializing these members to `undefined`. Won't this cause a de-opt if/when the members added outside of the ctor? And won't this code from `unsubscribe` involve a walking the prototype chain if the members are not present?

```ts
const { _parents, _singleParent, _teardowns } = this;
```

Reply:
Ah! Right you are! I was under the impression that TS would initialize them to `undefined` for us, but nope. Although, I wonder if "deopt"-wise if `null` is the proper choice. 🤔